src/test/java/org/apache/sling/i18n/impl/JcrResourceBundleProviderTest.java

```diff
@@ -112,4 +182,58 @@ public void testToLocaleWithPrivateUseCountryCode() {
         // Lowercase Private use Country 'xa'
         Assert.assertEquals(new Locale(Locale.GERMAN.getLanguage(), "XA"), JcrResourceBundleProvider.toLocale("de_xa"));
     }
+
+    @Test
+    public void testGetParentLocale() {
+        JcrResourceBundleProvider provider = new JcrResourceBundleProvider();
+        String variant = "variant1";
+        String script = "Hans";
+        // Locale with script and variant
+        Locale locale = new Locale.Builder()
+                .setLanguage(Locale.CHINA.getLanguage())
+                .setScript(script)
+                .setRegion(Locale.CHINA.getCountry())
+                .setVariant(variant)
+                .build();
+        Assert.assertEquals(
+                new Locale.Builder()
+                        .setLanguage(Locale.CHINA.getLanguage())
+                        .setScript(script)
+                        .setRegion(Locale.CHINA.getCountry())
+                        .build(),
+                provider.getParentLocale(locale));
+
+        // Locale with script and country
+        locale = new Locale.Builder()
+                .setLanguage(Locale.CHINA.getLanguage())
+                .setScript(script)
+                .setRegion(Locale.CHINA.getCountry())
+                .build();
+        Assert.assertEquals(
+                new Locale.Builder()
+                        .setLanguage(Locale.CHINA.getLanguage())
+                        .setScript(script)
+                        .build(),
+                provider.getParentLocale(locale));
+
+        // Locale with script only
+        locale = new Locale.Builder()
+                .setLanguage(Locale.CHINA.getLanguage())
+                .setScript(script)
+                .build();
+        Assert.assertEquals(new Locale(Locale.CHINA.getLanguage()), provider.getParentLocale(locale));
+
+        // Locale with variant only
+        locale = new Locale(Locale.CHINA.getLanguage(), Locale.CHINA.getCountry(), variant);
+        Assert.assertEquals(
+                new Locale(Locale.CHINA.getLanguage(), Locale.CHINA.getCountry()), provider.getParentLocale(locale));
+
+        // Locale with country only
+        locale = new Locale(Locale.CHINA.getLanguage(), Locale.CHINA.getCountry());
+        Assert.assertEquals(new Locale(Locale.CHINA.getLanguage()), provider.getParentLocale(locale));
+
+        // Locale with language only
+        locale = new Locale(Locale.CHINA.getLanguage());
+        Assert.assertEquals(provider.getDefaultLocale(), provider.getParentLocale(locale));
```

Comment:
To complete the coverage of the important (non-exception handling) branches I would add one additional validation.

```suggestion
        Assert.assertEquals(provider.getDefaultLocale(), provider.getParentLocale(locale));
        
         // The parent of the default locale is null
        Assert.assertNull(provider.getParentLocale(provider.getDefaultLocale())); 
```